src/gsi/setupfed.f90

```diff
@@ -521,17 +530,22 @@ subroutine setupfed(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,fed_diagsa
  ! 3) H(x), interpolate the FED from model space on the local domain to obs space (FEDMdiag)
  !============================================================================================
 
-         npt = 0
-         FEDMdiag(i) = 0.
-         call tintrp2a11(rp,FEDMdiag(i),dlat,dlon,dtime,hrdifsig,mype,nfldsig)
-         dlonobs(i) = dlon_earth
-         dlatobs(i) = dlat_earth
-
-         ! also Jacobian used for TLM and ADM
-         !FEDMdiagTL, used for gsi-3dvar,will be implemented in future......
-         FEDMdiagTL(i) = 0.
-         jqg_num = FEDMdiagTL(i)     !=dFED/Dqg
-         jqg  = jqg_num
+     npt = 0
+     FEDMdiag(i) = 0.
+     if (if_model_fed .and. innov_use_model_fed) then
```

Comment:
related to the comments by @Wangy1111 about the usage of innov_use_model_fed.   does if_model_fed not mean fed is not necessarily  in the background?

Reply:
@TingLei-NOAA 
Add the description. 
innov_use_model_fed: true, calculate OMB using FED reads in from Background (this needs if_model_fed=.true., fed in background). 
                                       false, calculate FED with hydrometers etal from background, and then calculate OMB.

Reply:
@TingLei-NOAA 
Do you have an idea where to put the check including the check to make sure the varialbe FED in anavinfo when EnVar is turned on.

Reply:
> OK, Then, I would suggest adding a check to make sure when innov_use_model_fed=.true., if_model_fed have to be .true.

@TingLei-NOAA 
done. Add check in gsimod.F90. 
  if (innov_use_model_fed .and. .not.if_model_fed) then
     if(mype==0) write(6,*)' GSIMOD: invalid innov_use_model_fed=.true. but if_model_fed=.false.'
     call die(myname_,'invalid if_model_fed, check namelist settings',335)
  end if